ADA/databricks_rstudio_sql_warehouse.qmd

```diff
@@ -1,105 +1,191 @@
+The Databricks token is a personal access token. 
 
-    i)  In the start menu, search 'ODBC' and open 'ODBC Data Sources (64-bit)'.
+A personal access token is is a security measure that acts as an identifier to let Databricks know who is accessing information from the SQL warehouse. Access tokens are usually set for a limited amount of time, so they will need renewing periodically.
 
-    ii) On the 'User DSN' tab click the 'Add...' button.
+-   In Databricks, click on your email address in the top right corner, then click 'User settings'
 
-    iii) In the 'Create New Data Source' window, select 'Simba Spark ODBC Driver' and click 'Finish'.
+-   Go to the 'Developer' tab in the side bar. Next to 'Access tokens', click the 'Manage' button
 
-    iv) In the 'Simba Spark ODBC Driver DSN Setup' window,
+::: {align="center"}
+![](../images/databricks-access-tokens.png)
+:::
 
-        a.  Enter a 'Data Source Name' and 'Description'. Choose a short and sensible data source name and note it down as this is what you will use to connect to Databricks through RStudio. *As you can set up more than one cluster on Databricks, use the description to make clear which cluster this connection is for. The description shown below describes that this connection is using an 8 core cluster on Databricks Runtime Environment 13.*
-        b.  Set the remaning options to the settings below.
-            -   Enter the 'Server Hostname' for your cluster in the 'Host(s):' field (you noted this down in step 2).
-            -   In the Port section, remove the default number and use the Port number you noted in step 2. 
-            -   Set the Authentication Mechanism to 'User Name and Password'.
-            -   Enter the word 'token' into the 'User Name:' field, then enter your 'Databricks access token' in the 'Password:' field.
-            -   Change the Thrift Transport option to HTTP.
-            -   Click the 'HTTP Options...' button and enter the 'HTTP Path' of your Databricks cluster, then click 'Okay'.
-            -   Click the 'SSL Options...' button and tick the 'Enable SSL' box, then click the 'OK' button.
-            -   ![](../images/odbc-driver-settings.png)
-        c.  Click the 'Test' button to verify the connection has worked. You should see the following message. *If you get an error here, repeat steps 5.e.i -- 5.e.ix again and ensure all the values are correct.*
+-   Click the 'Generate new token' button
 
-        ![](../images/databricks-test-connection.png)
+-   Name the token, then click 'Generate'
 
-        d.  Click the 'OK' button to exit the 'Test Results' window, then the 'OK' button in the 'Simba Spark ODBC Driver DSN Setup' window.
+::: callout-note
+Note that access tokens will only last as long as the value for the 'Lifetime (days)' field. After this period the token will expire, and you will need to create a new one to re-authenticate.
```

Comment:
Is there a max time or a recommended time you can set it for that we can include here?

Reply:
I can check this with the ADA team, I seem to remember it was discussed somewhere in the many comments above 😆